bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Executor.scala

```diff
@@ -0,0 +1,42 @@
+package edu.gemini.seqexec.server
+
+import edu.gemini.spModel.config2.{Config, ConfigSequence, ItemKey}
+import edu.gemini.spModel.obscomp.InstConstants.INSTRUMENT_NAME_PROP
+import edu.gemini.spModel.seqcomp.SeqConfigNames.INSTRUMENT_KEY
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+/**
+ * Created by jluhrs on 4/28/15.
+ */
+final class Executor(sequenceConfig: ConfigSequence) {
+
+  def run: Future[Int] = runSequence(sequenceConfig.getAllSteps.toList)
+
+  private def runSequence(config: List[Config]): Future[Int] = {
+    config match {
+      case Nil => Future.failed(new Exception("Empty sequence configuration"))
```

Comment:
I think an empty sequence might not be an error case?  If it is empty, it just "runs" immediately and can't fail.

At any rate, since we're just getting started on something new, I think it makes sense to switch to `Task` as Rob mentioned and use an application-defined failure type instead of exceptions.  Something like the `TryVcs` and `VcsAction` in the `vcs2` package object:

````scala
type TrySeq[A] = SeqFailure \/ A
type SeqAction[+A] = EitherT[Task, TrySeq, A]
````

A `Future` launches right away when it is created but this would allow us to decide when and how to run the effects later.